src/index.ts

```diff
@@ -515,6 +527,102 @@ export function create (rawOptions: CreateOptions = {}): Register {
   let getOutput: (code: string, fileName: string) => SourceOutput
   let getTypeInfo: (_code: string, _fileName: string, _position: number) => TypeInfo
 
+  const getCanonicalFileName = (ts as unknown as TSInternal).createGetCanonicalFileName(ts.sys.useCaseSensitiveFileNames)
```

Comment:
This seems dangerous, shouldn't case sensitivity be determined based on the OS, or potentially the [forceConsistentCasingInFileNames](https://www.typescriptlang.org/tsconfig#forceConsistentCasingInFileNames) option?

Reply:
Fair question.  `ts.sys.useCaseSensitiveFileNames` is OS-dependent: true on Posix, false on Windows.  As far as I can tell, `createGetCanonicalFileName` is how the compiler internally converts filenames to cache keys.

Reply:
As far as I know, `forceConsistentCasingInFileNames` generates additional semantic diagnostics but doesn't affect resolver behavior.  Even when `forceConsistentCasingInFileNames` is turned off, `FOO.ts` and `foo.ts` still resolve to the same file on Windows, right?

Reply:
Ah, that makes sense. I was reading `useCaseSensitiveFileNames` as if there was another `useCaseInsensitiveFileNames` option, which it makes sense that there isn't. This should be fine then.